fetch/api/resources/sw-intercept-abort.js

```diff
@@ -0,0 +1,20 @@
+async function messageClient(clientId, message) {
+  const client = await clients.get(clientId);
+  client.postMessage(message);
+}
+
+addEventListener('fetch', event => {
+  let resolve;
+  const promise = new Promise(r => resolve = r);
+
+  function onAborted() {
+     messageClient(event.clientId, event.request.signal.reason);
+     resolve();
+  }
+
+  messageClient(event.clientId, 'fetch event has arrived');
+
+  event.waitUntil(promise);
```

Comment:
Sorry I was wrong about this, we don't need `event.waitUntil`.

```js
event.respondWith(promise.then(() => new Response('hello'));
```

would be enough.

Reply:
Got it, thank you!